### PR TITLE
docs: Add query README file and explain some rationale

### PR DESCRIPTION
**Pull request by alamb**
This adds a README.md file in the query crate trying to capture the rationale and sweet sweet ASCII art. I figured that since @pauldix  said this matches what he currently has in mind  https://github.com/influxdata/influxdb_iox/pull/630#discussion_r555757382 I would get it into the repo.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Review by e-dard — DISMISSED** [review]

File: query/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+* Storage gRPC
+* Flux (possibly in the future)
+* InfluxQL (possibly in the future)
+* Others (possibly in the future)
+
+Sources of Chunk data
+* ReadBuffer
+* MutableBuffer
+* Parquet Files
+* Others (possibly in the fugure, like Remote Chunk?)
```
Comment (e-dard) [inline query/README.md:19]:
```suggestion
* Others (possibly in the future, like Remote Chunk?)
```

File: query/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+The goal is to use the shared query / plan representation in order to
+avoid N*M combinations of languge and Chunk source.
```
Comment (e-dard) [inline query/README.md:22]:
```suggestion
avoid N*M combinations of language and Chunk source.
```

File: query/README.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+Thus query planning is implemented in terms of traits, and those
+traits are implemented by different chunk impementations.
```
Comment (e-dard) [inline query/README.md:25]:
```suggestion
traits are implemented by different chunk implementations.
```

File: query/README.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+Amont othe rthings, this means that this crate should not depend
```
Comment (e-dard) [inline query/README.md:27]:
```suggestion
Among other things, this means that this crate should not depend
```

**Review by pauldix — APPROVED** [review]
looks great

**Review by e-dard — DISMISSED** [review]
Couple of typos otherwise LGTM 🥳

**Pull request merged**